imu_filter_madgwick/src/imu_filter_ros.cpp

```diff
@@ -308,86 +349,108 @@ void ImuFilterRos::publishTransform(const ImuMsg::ConstPtr& imu_msg_raw)
   rpy.vector.y = pitch;
-  rpy.vector.z = yaw ;
+  rpy.vector.z = yaw;
   rpy.header.stamp = t;
   rpy.header.frame_id = imu_frame_;
-  rpy_raw_debug_publisher_.publish(rpy);
+  rpy_raw_debug_publisher_->publish(rpy);
 }
 
-
-void ImuFilterRos::reconfigCallback(FilterConfig& config, uint32_t level)
+void ImuFilterMadgwickRos::reconfigCallback(const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
 {
   double gain, zeta;
-  boost::mutex::scoped_lock lock(mutex_);
-  gain = config.gain;
-  zeta = config.zeta;
-  filter_.setAlgorithmGain(gain);
-  filter_.setDriftBiasGain(zeta);
-  ROS_INFO("Imu filter gain set to %f", gain);
-  ROS_INFO("Gyro drift bias set to %f", zeta);
-  mag_bias_.x = config.mag_bias_x;
-  mag_bias_.y = config.mag_bias_y;
-  mag_bias_.z = config.mag_bias_z;
-  orientation_variance_ = config.orientation_stddev * config.orientation_stddev;
-  ROS_INFO("Magnetometer bias values: %f %f %f", mag_bias_.x, mag_bias_.y, mag_bias_.z);
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  for (auto &changed_parameter : event->changed_parameters)
+  {
+    const auto &type = changed_parameter.value.type;
+    const auto &name = changed_parameter.name;
+    const auto &value = changed_parameter.value;
+
+    if (type == ParameterType::PARAMETER_DOUBLE)
+    {
+      RCLCPP_INFO(get_logger(), "Parameter %s set to %f", name.c_str(), value.double_value);
+      if (name == "gain") {
+        gain = value.double_value;
+        filter_.setAlgorithmGain(gain);
+      } else if (name == "zeta") {
+        zeta = value.double_value;
+        filter_.setDriftBiasGain(zeta);
+      } else if (name == "mag_bias_x") {
+        mag_bias_.x = value.double_value;
+      } else if (name == "mag_bias_y") {
+        mag_bias_.y = value.double_value;
+      } else if (name == "mag_bias_z") {
+        mag_bias_.z = value.double_value;
+      } else if (name == "orientation_stddev") {
+        double orientation_stddev = value.double_value;
+        orientation_variance_ = orientation_stddev * orientation_stddev;
+      }
+    }
+  }
 }
 
-void ImuFilterRos::checkTopicsTimerCallback(const ros::TimerEvent&)
+void ImuFilterMadgwickRos::checkTopicsTimerCallback()
 {
   if (use_mag_)
-    ROS_WARN_STREAM("Still waiting for data on topics " << ros::names::resolve("imu") << "/data_raw"
-                    << " and " << ros::names::resolve("imu") << "/mag" << "...");
+    RCLCPP_WARN_STREAM(get_logger(), "Still waiting for data on topics "
+                                         << "~/imu/data_raw"
+                                         << " and "
+                                         << "~/imu/mag"
+                                         << "...");
   else
-    ROS_WARN_STREAM("Still waiting for data on topic " << ros::names::resolve("imu") << "/data_raw" << "...");
+    RCLCPP_WARN_STREAM(get_logger(), "Still waiting for data on topic "
+                                         << "~/imu/data_raw"
+                                         << "...");
```

Comment:
Don't forget to remove the `"~/"` here as well. Also, these strings can now be merged.